frontend/src/services/protectedemail.service.ts

```diff
@@ -0,0 +1,30 @@
+export interface CsvStatus {
+  csvFilename?: string
+  csvError?: string
+  numRecipients?: number
+  preview?: string
+}
+
+export async function extractTemplateParams(
+  template: string
+): Promise<Array<string>> {
+  return Promise.resolve(['recipient', 'password', 'secretmsg'])
+}
+
+export async function validateCsv(
```

Comment:
Hmmmm I didn't know we were going to do this, but it does seem like a good idea if we don't mind parsing the file twice. 

What I mean is that in this function, we could parse the whole file using `papaparse` and make sure that `recipient`, `password` and any params used in the template is in each row of the csv and that there are no parsing related errors.

If there is any error in this stage, we will not proceed to encrypt the content, hash the password, or upload the file.

Once we are sure that the params have no errors, we could proceed with the uploading. I have written some thoughts on what the uploading would be like. #364